magefile.go

```diff
@@ -104,3 +124,23 @@ func Precommit() error {
 func Check() {
 	mg.SerialDeps(Lint, Test)
 }
+
+// BuildExamples builds all the examples
+func BuildExamples() error {
```

Comment:
I don't know if we really need this given we have the tests, we're not going to publish example binaries, so just running tests should be equivalent to building them in terms of verifying the code

Reply:
The idea was to have a first broader check on any (potential) example taking into account also (potential) examples that may not have tests. But I see, too many "potential" in a sentence 😅

Reply:
In general I think it is a good idea to verify examples are buildable because that is probably one of the first things people with interact with.

Reply:
Yeah it's why we need tests to make sure they not only build but also work too :) So since we are running the tests, I don't think we need the build, we shouldn't add examples ever without any tests to make sure they aren't broken and leading users into a rabbit hole.

Reply:
Yeah, I mean my point is, let's have both as it is free :p. If you are very inclined to not to build them I am fine.

Reply:
I'm fairly inclined ;) I think over time or causes confusion if we do the same thing twice, which is what happens for these when running both go test and go build I think